### PR TITLE
[SYCL] Replaces some of the CL_* enums with PI_* enums.

### PR DESCRIPTION
**Pull request by rbegam**
Signed-off-by: rbegam <rehana.begam@intel.com>

Co-Authored-By: Alexey Bader <alexey.bader@intel.com>

### CODE REVIEW

**Comment by bader** [comment]
@rbegam, please, resolve merge conflict and clang-format check issues.

**Comment by bader** [comment]
@rbegam, please, resolve merge conflicts.

**Comment by rbegam** [comment]
@bader done. please review.

File: sycl/include/CL/sycl/queue.hpp [hunk 1]
```diff
@@ -286,11 +286,11 @@ class queue {
...
   ///
-  /// \param Ptr is a USM pointer to the allocation.
-  /// \param Length is a number of bytes in the allocation.
-  /// \param Advice is a device-defined advice for the specified allocation.
-  /// \return an event representing advice operation.
-  event mem_advise(const void *Ptr, size_t Length, int Advice);
+  /// @param Ptr is a USM pointer to the allocation.
+  /// @param Length is a number of bytes in the allocation.
+  /// @param Advice is a device-defined advice for the specified allocation.
+  /// @return an event representing advice operation.
```
Comment (bader) [inline sycl/include/CL/sycl/queue.hpp]: Please, revert these changes.

File: sycl/source/detail/queue_impl.hpp [hunk 2]
```diff
@@ -333,10 +333,10 @@ class queue_impl {
   /// Provides additional information to the underlying runtime about how
   /// different allocations are used.
   ///
-  /// \param Ptr is a USM pointer to the allocation.
-  /// \param Length is a number of bytes in the allocation.
-  /// \param Advice is a device-defined advice for the specified allocation.
-  event mem_advise(const void *Ptr, size_t Length, int Advice);
+  /// @param Ptr is a USM pointer to the allocation.
+  /// @param Length is a number of bytes in the allocation.
+  /// @param Advice is a device-defined advice for the specified allocation.
```
Comment (bader) [inline sycl/source/detail/queue_impl.hpp]: Please, revert these changes.

File: sycl/include/CL/sycl/detail/pi.h [hunk 3]
```diff
@@ -389,25 +464,45 @@ const pi_queue_properties PI_QUEUE_ON_DEVICE_DEFAULT =
...
-typedef _pi_command_type pi_command_type;
-typedef _pi_program_info pi_program_info;
 typedef _pi_program_build_info pi_program_build_info;
 typedef _pi_program_build_status pi_program_build_status;
 typedef _pi_kernel_info pi_kernel_info;
-typedef _pi_kernel_group_info pi_kernel_group_info;
+
+// For compatibility with OpenCL define this not as enum.
+using pi_device_partition_property = intptr_t;
+static constexpr pi_device_partition_property PI_DEVICE_PARTITION_EQUALLY =
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.h]: @smaslov-intel, can we use C++ in this header?
If so, I suggest keep the coding style consistent and replace all `typedef` with `using`.
Why we have two separate files: pi.h and pi.hpp?

File: sycl/include/CL/sycl/queue.hpp (hunk 1 above)
Comment (rbegam) [reply sycl/include/CL/sycl/queue.hpp]: @bader pi_mem_advice is introduced in this patch. Reverting this in this file will require removing it altogether. Is that what we want?

File: sycl/source/detail/queue_impl.hpp (hunk 2 above)
Comment (rbegam) [reply sycl/source/detail/queue_impl.hpp]: same as above.

File: sycl/include/CL/sycl/queue.hpp (hunk 1 above)
Comment (bader) [reply sycl/include/CL/sycl/queue.hpp]: This request is about comments only.

Comment (rbegam) [reply sycl/include/CL/sycl/queue.hpp]: that makes sense. on it.

File: sycl/include/CL/sycl/detail/pi.h (hunk 3 above)
Comment (rbegam) [reply sycl/include/CL/sycl/detail/pi.h]: done.

**Comment by rbegam** [comment]
@romanovvlad all comments have been resolved. please submit as I do not have write permission. thanks.

File: sycl/include/CL/sycl/detail/pi.h [hunk 4]
```diff
@@ -86,16 +86,6 @@ typedef enum {
...
 
-typedef enum {
-  PI_COMMAND_KERNEL_LAUNCH = CL_COMMAND_NDRANGE_KERNEL,
-  PI_COMMAND_MEMBUFFER_WRITE = CL_COMMAND_WRITE_BUFFER,
-  PI_COMMAND_MEMBUFFER_READ = CL_COMMAND_READ_BUFFER,
-  PI_COMMAND_USER = CL_COMMAND_USER,
-  PI_COMMAND_EVENTS_WAIT = CL_COMMAND_MARKER,
-  PI_COMMAND_MEMBUFFER_COPY = CL_COMMAND_COPY_BUFFER,
-  PI_COMMAND_MEMBUFFER_FILL = CL_COMMAND_FILL_BUFFER
-} _pi_command_type;
```
Comment (fwyzard) [inline sycl/include/CL/sycl/detail/pi.h]: These changes break the CUDA plugin, e.g. https://github.com/intel/llvm/blob/dc729a72b5bd815ffd7e118cd8d1992c5248ee4a/sycl/plugins/cuda/pi_cuda.cpp#L1634 *et al.*

Comment (rbegam) [reply sycl/include/CL/sycl/detail/pi.h]: Those are renamed i.e., PI_COMMAND_MEMBUFFER_WRITE -> PI_COMMAND_TYPE_MEM_BUFFER_WRITE

Comment (fwyzard) [reply sycl/include/CL/sycl/detail/pi.h]: Yes, but the renaming was not propagated to the CUDA plugin.

Comment (bader) [reply sycl/include/CL/sycl/detail/pi.h]: @fwyzard, https://github.com/intel/llvm/pull/1310 should fix that issue.

Comment (fwyzard) [reply sycl/include/CL/sycl/detail/pi.h]: Thanks, that worked.

Comment (rbegam) [reply sycl/include/CL/sycl/detail/pi.h]: @fwyzard sorry about that. Why wasn't there any indication in the tests that it broke CUDA plugin? Was I needed to run some other tests by myself? @bader

Comment (bader) [reply sycl/include/CL/sycl/detail/pi.h]: > @fwyzard sorry about that. Why wasn't there any indication in the tests that it broke CUDA plugin? Was I needed to run some other tests by myself?

+@pvchupin to reply

Comment (pvchupin) [reply sycl/include/CL/sycl/detail/pi.h]: There is no buildbot to check that at the moment, but we are working on adding it.
Meanwhile to check locally you can try this https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain-with-support-for-nvidia-cuda
